x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2867 @@
+{
+  name: "Original name",
+  rule_source: {
+    type: "external",
+    is_customized: false,
+    ...
+  }
+}
+</pre>
+    </td>
+    <td>
+      <pre>
+{
+  name: "Original name",
+}</pre>
+    </td>
+    <td>
+      <pre>
+{
+  name: "My custom name",
+}</pre>
+    </td>
+    <td>
+<pre>
+{
+  name: "My custom name",
+  rule_source: {
+    type: "external",
+    is_customized: true,
+    ...
+  }
+}
+</pre>
+    </td>
+  </tr>
+
+
+
+  
+  <tr>
+    <td>New field added, not existing in base asset</td>
+    <td>
+<pre>
+{
+  name: "Original name",
+  rule_source: {
+    type: "external",
+    is_customized: false,
+    ...
+  }
+}
+</pre>
+    </td>
+    <td>
+      <pre>
+{
+  name: "Original name",
+}</pre>
+    </td>
+    <td>
+      <pre>
+{
+  name: "Original name",
+  note: "My investigation guide",
+}</pre>
+    </td>
+    <td>
+<pre>
+{
+  name: "Original name",
+  note: "My investigation guide",
+  rule_source: {
+    type: "external",
+    is_customized: true,
+    ...
+  }
+}
+</pre>
+    </td>
+  </tr>
+
+
+
+
+
+
+  
+  <tr>
+    <td>Modified matches value in base asset</td>
+    <td>
+<pre>
+{
+  name: "My custom name",
+  rule_source: {
+    type: "external",
+    is_customized: true,
+    ...
+  }
+}
+</pre>
+    </td>
+    <td>
+      <pre>
+{
+  name: "Original name",
+}</pre>
+    </td>
+    <td>
+      <pre>
+{
+  name: "Original name",
+}</pre>
+    </td>
+    <td>
+<pre>
+{
+  name: "Original name",
+  rule_source: {
+    type: "external",
+    is_customized: false,
+    ...
+  }
+}
+</pre>
+    </td>
+  </tr>
+
+
+
+
+
+
+  
+  <tr>
+    <td>Field existing in base asset is removed</td>
+    <td>
+<pre>
+{
+  name: "Original name",
+  "tags": ["Linux", "Windows"]
+  rule_source: {
+    type: "external",
+    is_customized: false,
+    ...
+  }
+}
+</pre>
+    </td>
+    <td>
+      <pre>
+{
+  name: "Original name",
+  "tags": ["Linux", "Windows"]
+}
+      </pre>
+    </td>
+    <td>
+      <pre>
+{
+  name: "Original name",
+}
+      </pre>
+    </td>
+    <td>
+<pre>
+{
+  name: "Original name",
+  rule_source: {
+    type: "external",
+    is_customized: true,
+    ...
+  }
+}
+</pre>
+    </td>
+  </tr>
+
+
+
+
+
+
+
+
+  
+  <tr>
+    <td>Edge case: base asset not found</td>
+    <td>
+<pre>
+{
+  name: "Original name",
+  rule_source: {
+    type: "external",
+    is_customized: false,
+    ...
+  }
+}
+</pre>
+    </td>
+    <td>
+      <center><b>Not found</b></center>
+    </td>
+    <td>
+      <pre>
+{
+  name: "Original name",
+}</pre>
+    </td>
+    <td>
+<pre>
+{
+  name: "Original name",
+  rule_source: {
+    type: "external",
+    is_customized: true,
+    ...
+  }
+}
+</pre>
+    </td>
+  </tr>
+
+
+
+
+
+  </tbody>
+  </table>
+
+Notice the last scenario from the table, which deals with the edge case of the corresponding version of the `security-rule` asset not being found. This would prevent from doing a comparison of the fields to calculate if the rule should be considered customized or not.
```

Comment:
Could you provide a few examples why `security-rule` asset isn'f found? Obviously it could be a problem with prebuilt rules installation or manually removing the asset. But will each rule update endpoint wait for prebuilt rules to be installed or not?

Reply:
> Could you provide a few examples why security-rule asset isn'f found?

- User manually deletes package and assets
- A user imports a prebuilt rule with matching `rule_id` but no matching `version` (base version missing)

> But what if the user then manually deleted the package, and all the assets were deleted as well?

I think that the behaviour should be to still mark the rule as `is_customized: true`, given the reasons provided below. Do you have in mind any reasons why it should behave differently?

Reply:
> I think that the behaviour should be to still mark the rule as is_customized: true, given the reasons provided below.

@jpdjere This makes sense to me.

Reply:
TODO:
- Add edge case of importing rule from newer version of Kibana/rules package to older version, and the base version not being found cause it doesn't exist yet. -> keep the `version` from payload and mark as `is_customized: true`.